config/webhook/manifests.yaml

```diff
@@ -0,0 +1,56 @@
+
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  creationTimestamp: null
+  name: mutating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /mutate-managed-gitops-redhat-com-v1alpha1-gitopsdeployment
+  failurePolicy: Fail
+  name: mgitopsdeployment.kb.io
+  rules:
+  - apiGroups:
+    - managed-gitops.redhat.com
+    apiVersions:
+    - v1alpha1
+    operations:
+    - CREATE
+    - UPDATE
+    resources:
+    - gitopsdeployments
+  sideEffects: None
+
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  creationTimestamp: null
+  name: validating-webhook-configuration
+webhooks:
+- admissionReviewVersions:
+  - v1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /validate-managed-gitops-redhat-com-v1alpha1-gitopsdeployment
+  failurePolicy: Fail
+  name: vgitopsdeployment.kb.io
+  rules:
+  - apiGroups:
+    - managed-gitops.redhat.com
```

Comment:
@ishitasequeira @anandf I don't think it's a good idea to introduce Red Hat managed services specific configuration here in argocd-operator 

I think if this is required we should think of a way to do this in gitops-operator instead